### ISSUE TITLE
[30k]: Search UI

### ISSUE THREAD

**Issue by widal001** [HHS/simpler-grants-gov#89]
### Key links

- [Deliverable specification](https://simpler-grants.gitbook.io/wiki/product/deliverables/specifications/search-ui)
- [Simpler Grants.gov full roadmap](https://github.com/orgs/HHS/projects/12)
- [10k milestones](https://github.com/orgs/HHS/projects/13/views/11?sliceBy%5Bvalue%5D=Search+ui)

### Description

- **What:** Release a basic search page on simpler.grants.gov, potentially hidden behind a feature flag, that allows users to filter and sort funding opportunities, then save and share the link to the results page
- **Why:** Proves the successful integration of front-end and backend systems and delivers basic search functionality for user testing
- **Who:** HHS leadership and targeted group of internal and external stakeholders with whom we'd like to conduct user research and testing

> [!NOTE]
> For more information about this deliverable please refer to the deliverable specification linked above

**Comment by sarahknoppA6** [comment]
@sumiat @widal001 Team suggested a v0 without comms and then a separate 30k for search v1 (with comms)

**Comment by EOKENAVA** [comment]
Weekly Update

What has been done?

Reviewed draft Figma UI with MH and HHS.

Status
On Track

Risk
None

Deliverable Date
TBD

What will be done?

- Approval of the Figma UI with MH and Emily
- Service blueprint of the grants application process to inform search
- Start to create testing brief (might be dependent on approval)
- Set up the plumbing/flow for API calls
- Mock data set up

**Comment by lucasmbrown-usds** [comment]
Will the testing brief include our plan for user research, usability testing, etc?

**Comment by andycochran** [comment]
> Will the testing brief include our plan for user research, usability testing, etc?

Yeah. We'll first create a research brief to define what/how/who ([template](https://app.gitbook.com/o/cFcvhi6d0nlLyH2VzVgn/s/v1V0jIH7mb7Yb3jlNrgk/design/design-research/research-documentation/template-research-brief-summary)), then a testing plan with a methodology, timeline, script ([template](https://app.gitbook.com/o/cFcvhi6d0nlLyH2VzVgn/s/v1V0jIH7mb7Yb3jlNrgk/design/design-research/research-documentation/template-usability-test-plan-summary)).

**Comment by EOKENAVA** [comment]
Updated Delivery Date - 5/14

**Comment by andycochran** [comment]
[v7 Figma](https://www.figma.com/file/Zm7oIeBpwjrsXcmjV3K7HR/Search?type=design&node-id=4977-1764&mode=design&t=X67wUo796tzdrzW2-0) reviewed was just with OG Leadership; no changes requested; ready for development

**Comment by EOKENAVA** [comment]
**Weekly Update**

**What has been done?**
- Received approval of FIGMA from OG Leadership with no additional changes.
- Collected information for service blueprint of grants.gov grantor posting process

**Status**
On Track

**Risk**
None

**Deliverable Date**
5/14

**What will be done?**

- Develop responsive search input with placeholder text and button.
- Create a research brief ([template](https://app.gitbook.com/o/cFcvhi6d0nlLyH2VzVgn/s/v1V0jIH7mb7Yb3jlNrgk/design/design-research/research-documentation/template-research-brief-summary)) that breaks down the design work to be done, the business motivations, and general effort/timeline.

**Comment by EOKENAVA** [comment]
**Weekly Update**

**What has been done?**
- Set up component/markup for Funding Instrument filter
- HHS Approved Search UI Figma 
- Created a component that takes the result data returned by the api and displays the results components in a list.

**Status**
On Track

**Risk**
None

**Deliverable Date**
5/14

**What will be done?**
- Create a research brief ([template](https://app.gitbook.com/o/cFcvhi6d0nlLyH2VzVgn/s/v1V0jIH7mb7Yb3jlNrgk/design/design-research/research-documentation/template-research-brief-summary)) that breaks down the design work to be done, the business motivations, and general effort/timeline.
- Investigate roadmap for opportunities for user research

**Comment by margaretspring** [comment]
**Weekly Update**

**What has been done?**
- Set up component/markup for Funding Instrument filter
- HHS Approved Search UI Figma 
- Created a component that takes the result data returned by the api and displays the results components in a list.

**Status**
On Track

**Risk**
None

**Deliverable Date**
5/14

**What will be done?**
- Create a research brief ([template](https://app.gitbook.com/o/cFcvhi6d0nlLyH2VzVgn/s/v1V0jIH7mb7Yb3jlNrgk/design/design-research/research-documentation/template-research-brief-summary)) that breaks down the design work to be done, the business motivations, and general effort/timeline.
- Investigate roadmap for opportunities for user research

**Comment by margaretspring** [comment]
**Weekly Update - IN DRAFT**

**What has been done?**
-
-

**Status**


**Risk**


**Deliverable Date**


**What will be done?**
-

**Comment by hhs-swift** [comment]
@widal001 @margaretspring @sarahknoppA6  -- I'm noticing that the metrics from GitBook aren't in the acceptance criteria of this GitHub ticket (https://app.gitbook.com/o/cFcvhi6d0nlLyH2VzVgn/s/Pm7UEzeiS1tbLCV1SFRu/product/deliverables/specifications/search-user-interface/~/comments/YEV8iAqqa38I45d9Y4gc?node=d36a16da675d4a6dbe59fc8c555e5726#proposed-metrics). 

Do we want to copy those over, or is it fine that we don't have them here? What's our best practice for making sure they get implemented by the 30k date of 5/14?

**Comment by lucasmbrown-usds** [comment]
@widal001 @margaretspring @sarahknoppA6 -- I'm noticing that the metrics from GitBook aren't in the acceptance criteria of this GitHub ticket (https://app.gitbook.com/o/cFcvhi6d0nlLyH2VzVgn/s/Pm7UEzeiS1tbLCV1SFRu/product/deliverables/specifications/search-user-interface/~/comments/YEV8iAqqa38I45d9Y4gc?node=d36a16da675d4a6dbe59fc8c555e5726#proposed-metrics).

Do we want to copy those over, or is it fine that we don't have them here? What's our best practice for making sure they get implemented by the 30k date of 5/14?

**Comment by andycochran** [comment]
**Weekly Update**

**What has been done?**
- In-progress search page demo'd at ToT
- Testing plan drafted (still in internal review) 
- [ @acouch / @rylew1 / @chouinar  — please add any dev tasks that have happened since the last update ]
- [ @acouch is there an update on SIA for Ethnio? ] 

**Status**
In progress

**Risk**
- (Same as last update) Ethnio procurement and SIA will impact the progression of this work — We can proceed with usability tests with internal SME and grantors, but Ethnio blocks further tests with applicants 

**Deliverable Date**
5/14

**What will be done?**
- Design: usability testing plan to be reviewed by MicroHealth then sent to HHS
- Error states
- Set up environment vars for dev, stage, and prod

**Comment by andycochran** [comment]
**Weekly Update 4/10**

* [[10k] Search UI - Create initial search page](https://github.com/HHS/simpler-grants-gov/milestone/134)
  * **status** 72 points initially estimated, 100.1 pointed, 84.1 completed
  * completed
    * Update Next to 14.1 
    * Add eligibility and category filters
    * Frontend tests for search components
* [[10k] Search UI – Design and usability testing](https://github.com/HHS/simpler-grants-gov/milestone/142)
  * **status** 8 points initially estimated, 33 pointed, 22 completed 
  * blocked
* [[10k] Search UI - Analytics](https://github.com/HHS/simpler-grants-gov/milestone/147)
  * **status** 29 points initially estimated, 3 pointed, 3 completed 
  * completed 
    * n/a
* [[10k] Search UI - Feature flags, API, and infrastructure](https://github.com/HHS/simpler-grants-gov/milestone/135)
  * **status** 29 points initially estimated, 18 pointed, 14 completed 
  * completed
    * Setup VSCode API Debugger 

**What has been done?**
- Continued work on UI components, filters, and tests 
- Usability testing plan still in internal review 
- Mentioned at onsite that MH would work on setting up GA4 tracking on g.gov per proposal sent

**Status**
In progress

**Risk**
- Usability testing at risk of not beginning by 5/14, blocked by: 
  - Incentives 
  - Ethnio SIA 
  - API work / getting data into an environment that can be used for testing
  - (Note: these do not necessarily block usability tests with internal SMEs)

**Deliverable Date**
5/14

**What will be done?**
- Design: 
  - Explore alternative paths for user research tooling 
  - Usability testing plan review 
- Eng: 
  - Collapse sidebar/filters on small screens (mobile view)
  - Set up all agencies as filter options
  - Add USWDS error/alert when selecting a page beyond the max page possible
  - Setup i18n with next-intl
  - Search Results Loading indicator
  - Search error state for 400 and 500 errors
  - Frontend tests for server components (search page, page not found)
  - Set up environment vars for dev, stage, and prod
- Product:
  - Create tickets to set up tracking for [site analytics](https://github.com/HHS/simpler-grants-gov/milestone/147)

**Comment by widal001** [comment]
Updates ticket format to reduce discrepancies with the deliverable spec.

**Comment by margaretspring** [comment]
**Weekly Update 4/17**

**What has been done?**


**Status**


**Risk**
  

**Deliverable Date**


**What will be done?**

**Comment by margaretspring** [comment]
**Weekly Update 4/24**

**What has been done?**



**Status**


**Risk**


**Deliverable Date**


**What will be done?**

**Comment by lucasmbrown-usds** [comment]
When we say "UI will be done this sprint", do we mean we'll have a fully working search experience in the UI in PROD by the end of this sprint (May 1st)?

**Comment by andycochran** [comment]
Meant the DoDs will be satisfied this sprint. There are some details in the Design/Figma that may be a little rough, and we plan to address in the fast-follow milestone. But functionally, that's what we mean saying the "UI will be done." Please excuse the fuzzy term; as we know, nothing's ever _done_. 

Since the last update was posted, I've started QAing in lowers (just now available today). We're checking to make sure the UI is acceptable for usability testing. Once sure of that, we'll start sending out recruiting emails (likely Monday). 

@acouch can you confirm this sprint'll include deploying to prod?

**Comment by lucasmbrown-usds** [comment]
Sounds good, thanks Andy. Totally understand there will still be some bits and pieces to be worked out. 

I'm definitely interested in the PROD deploy, since as we all know, going to PROD always introduces new unexpected wrinkles, and you're never really sure whether things will work in PROD until they're in PROD!

**Comment by acouch** [comment]
@lucasmbrown-usds agreed about things being in production. This 30K is the UI side of the search page, so the scope of it is focused on just the frontend. The working code is in prod, but it won't show any results until the #972 transformation work is completed, which is scheduled for next sprint. It makes sense to not call this finished until the API work is done.

**Comment by lucasmbrown-usds** [comment]
Makes sense -- thanks @acouch !

**Comment by margaretspring** [comment]
**Weekly Update 5/1**

**What has been done?**
- 

**Status**
In Progress

**Risk**
 

**Deliverable Date**


**What will be done?**

**Comment by sarahknoppA6** [comment]
Development team has updated anticipated Delivery Date to 6/11 given additional work to improve the data model. https://betagrantsgov.slack.com/archives/C06F55BKLKX/p1714684814304889

**Comment by acouch** [comment]
**Weekly Update 5/8**

**What has been done?**
- Bug fixing and technical debt work
- GA4 setup in dev

**Status**
- In Progress

**Risk**
- #1532 
- Delivery depends on the transformation work 

**Deliverable Date**
- 6/11

**What will be done?**
- GA4 staging

**Comment by widal001** [comment]
Replacing our old deliverable tickets with a new ticket format

**Comment by HartleySan** [comment]
Hello. I came across this thread via https://simpler.grants.gov/process, and I'm curious what the status of this API is. Is it usable yet?

We're actively looking for a way to search through US government grants via an API, and this seems like a very promising option, but we can't tell whether it's actually available yet, or actively being worked on, or it will never be completed. Could you please share a status update?

Thank you.

**Comment by btabaska** [comment]
> Hello. I came across this thread via https://simpler.grants.gov/process, and I'm curious what the status of this API is. Is it usable yet?
> 
> We're actively looking for a way to search through US government grants via an API, and this seems like a very promising option, but we can't tell whether it's actually available yet, or actively being worked on, or it will never be completed. Could you please share a status update?
> 
> Thank you.

Hello,

Thanks for reaching out to us. This is actively being worked on as a federally funded Dept. of HHS effort and we are currently in early days as we build out the future of the federal grants experience. 

Yes this is available! in a closed alpha test as we begin to roll out the v1 of the API. I do want to note that this version will be deprecated over the coming months as we push improvements, but we will give users plenty of notice beforehand. If you are interested in getting an API key please reach out to us via email at simpler@grants.gov or by email to myself at brandontabaska@navapbc.com.
